### ISSUE TITLE
How to I convert older code to the new version??

### ISSUE THREAD

**Issue by rmControls** [letsar/flutter_staggered_grid_view#217]
Hello @letsar,

Congratulations on the updated package... 

after updating the package I am getting error with below code... can you please help me to covert the below example code with the working one?? 

```dart
SliverStaggeredGrid.countBuilder(
                crossAxisCount: 2,
                itemCount: mylist.length,
                mainAxisSpacing: 8,
                crossAxisSpacing: 8,

                itemBuilder: (context, index) {
                  final itemData= mylist[index];
                  return ImgTile(
                    title: itemData.title,
                    price: itemData.price!.toStringAsFixed(0),
                    imageUrl: itemData.image,
                    itemId: itemData.itemId,
                    onTap: () {
                    // some function
                    },
                  );
                },
                staggeredTileBuilder: (index) => StaggeredTile.fit(1),
                // staggeredTileBuilder: (index) => StaggeredTile.count(1, 1),
              ),
```

I searched in changelog or in example code in readme to see if we have any suggestion/code for converting from old package to the new. But unfortunately I didn't find any helpful info..

**Comment by saadbinzia** [comment]
I'm facing the same issue

**Comment by letsar** [comment]
The `SliverStaggeredGrid` has been removed, but for your use case, you certainly want to use the `SliverMasonryGrid`:

```dart
SliverMasonryGrid.count(
  crossAxisCount: 2,
  mainAxisSpacing: 8,
  crossAxisSpacing: 8,
  itemBuilder: (context, index) {
    final itemData = mylist[index];
    return ImgTile(
      title: itemData.title,
      price: itemData.price!.toStringAsFixed(0),
      imageUrl: itemData.image,
      itemId: itemData.itemId,
      onTap: () {
        // some function
      },
    );
  },
  childCount: mylist.length,
),
```

**Comment by ciamax** [comment]
Same problem here. How to convert this code?
`
    StaggeredGridView.countBuilder(
      crossAxisCount: _curOsservazione!.getValue('uicolumns')?? 4,
      controller: _scrollController,
      itemCount: staggeredTiles.length,
      staggeredTileBuilder: (index) {
        return staggeredTiles[index];
      },
      itemBuilder: (BuildContext context, int index) {
        //return uiElements[index];
        return AutoScrollTag(
          key: ValueKey(index),
          controller: _scrollController!,
          index: index,
          child: uiElements[index],
        );
      },
      mainAxisSpacing: cardSpacing,
      crossAxisSpacing: cardSpacing,
    );`
`

**Comment by letsar** [comment]
@ciamax it depends on which layout you want.
Does you layout look like this ?
![image](https://user-images.githubusercontent.com/9378033/147738942-8b1557a1-9d3a-4bc6-b1b8-af9f2d17802f.png)
If it's the case you can use the `StaggeredGrid` but it's only intented for a few items.

**Comment by ciamax** [comment]
> @ciamax it depends on which layout you want. Does you layout look like this ? ![image](https://user-images.githubusercontent.com/9378033/147738942-8b1557a1-9d3a-4bc6-b1b8-af9f2d17802f.png) If it's the case you can use the `StaggeredGrid` but it's only intented for a few items.

Yes that's the case but the new library seems to lack StaggeredTile.fit() (autofitting tiles) and the controller property for custom scroll.

**Comment by letsar** [comment]
For the controller is not a problem since you can wrap the `StaggeredGrid` with a `SingleChildScrollView`, but yes, I did not add the `fit` constructor in this version. What is the use case where you need it?

**Comment by ciamax** [comment]
It's just a simple case in which I need the tile to adjust its height dynamically depending on content

**Comment by letsar** [comment]
@ciamax I added the fit constructor on the master branch. Can you test it? If it's ok I'll publish this version.

**Comment by ciamax** [comment]
@letsar I made some tests and the fit constructor seems to work. It also works wrapping `StaggeredGrid.count()` in a `SingleChildScrollView` to get control over scrolling.

Super! :+1:

**Comment by letsar** [comment]
Published in 0.5.1, I'm closing it then.

**Comment by zae1403** [comment]
> @ciamax I added the fit constructor on the master branch. Can you test it? If it's ok I'll publish this version.

how to use StaggeredGridTile.fit with SliverMasonryGrid?
